### ISSUE TITLE
SMC sampling from prior stop working

### ISSUE THREAD

**Issue by aloctavodia** [pymc-devs/pymc#3025]
Before #2984 this used to work

```python
with pm.Model() as m:
    alfa = pm.HalfNormal('alfa', 20)
    p = pm.Beta('p', alfa, 1)
pm.distributions.draw_values([m['p']], size=10)
```
Also a model like this will fail when sampled using SMC, because SMC is initialized by sampling from the prior.

**Comment by ColCarroll** [comment]
Shoot.  Hoping to land something this week that will fix this and some
other shape problems.

**Comment by aloctavodia** [comment]
Great! Thank you!

**Comment by ColCarroll** [comment]
(got to my computer with the PR on it) 

The output is (something like)

```
[array([0.98216117, 0.99751651, 0.98250691, 0.97701074, 0.87199045,
        0.08888242, 0.93154041, 0.88219556, 0.99708625, 0.81510828])]
```
now.  Let me know if that looks wrong!

**Comment by aloctavodia** [comment]
That looks like a reasonable output.

**Comment by ColCarroll** [comment]
Fixed by #2983 (output is similar to above on master).
